### PR TITLE
update misc tools

### PR DESCRIPTION
**Pull request by jacob-delgado**

### CODE REVIEW

File: docker/build-tools/Dockerfile [hunk 1]
```diff
@@ -68,11 +68,11 @@ ENV SU_EXEC_VERSION=0.2
 ENV YQ_VERSION=4.31.1
 ENV KPT_VERSION=v0.39.3
 ENV BUF_VERSION=v1.13.1
-ENV GCLOUD_VERSION=416.0.0
+ENV GCLOUD_VERSION=420.0.0
 ENV KUBETEST2_VERSION=b019714a389563c9a788f119f801520d059b6533
-ENV COSIGN_VERSION=v1.13.1
+ENV COSIGN_VERSION=v2.0.0
```
Comment (howardjohn) [inline docker/build-tools/Dockerfile:73]: any breaking changes?

**Review by howardjohn — APPROVED** [review]

File: docker/build-tools/Dockerfile (hunk 1 above)
Comment (jacob-delgado) [reply docker/build-tools/Dockerfile:73]: https://github.com/sigstore/cosign/blob/main/CHANGELOG.md

Yes, but we should NOT merge in common-files PR until I see if i can get through the changes in release-builder

Comment (ericvn) [reply docker/build-tools/Dockerfile:73]: I looked at the changelog yesterday and what we do in release-builder and it seems we should be OK. But sounds like @jacob-delgado will test as well.

**Comment by ericvn** [comment]
kubectl has a new version as well.

**Pull request merged**